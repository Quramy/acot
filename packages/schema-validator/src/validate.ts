import { validate as validateSchema, ValidationError } from 'schema-utils';
import type { Schema } from './types';

export { ValidationError };

export type ValidateConfig = {
  name: string;
  base: string;
};

export const validate = <T = unknown>(
  schema: Schema,
  input: unknown,
  config: ValidateConfig,
): input is T => {
  validateSchema(schema, input as any, {
    name: config.name,
    baseDataPath: config.base,
  });

  return true;
};
