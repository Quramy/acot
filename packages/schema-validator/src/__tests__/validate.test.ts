import { validate } from '../validate';

describe('validate', () => {
  test('call schema-utils validate', () => {
    expect(() => {
      validate(
        {
          properties: {
            foo: {
              type: 'array',
              items: {
                type: 'number',
              },
            },
            bar: {
              type: 'string',
            },
          },
          additionalProperties: false,
        },
        {
          foo: ['string', 'test'],
          bar: null,
          baz: false,
        },
        {
          name: 'Mock',
          base: 'options',
        },
      );
    }).toThrowErrorMatchingSnapshot();
  });
});
