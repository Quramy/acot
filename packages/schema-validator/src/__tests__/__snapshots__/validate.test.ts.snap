// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate call schema-utils validate 1`] = `
"Invalid options object. Mock has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'baz' | should be any non-object. These properties are valid:
   object { foo?, bar? }
 - options.foo[0] should be a number.
 - options.foo[1] should be a number.
 - options.bar should be a string."
`;
